### PR TITLE
fix mach spawn fail interrupt

### PR DESCRIPTION
**Pull request by vtjnash**
I believe this should fix the occasional failure we've seen this past year on the spawn test on macOS where it gets a spurious InterruptException generated by the mach kernel. I guess that's what the man page meant when it said this function could have "unspecified behavior".

### CODE REVIEW

File: deps/libuv.version [hunk 1]
```diff
@@ -1,2 +1,2 @@
 LIBUV_BRANCH=julia-uv1.9.0
-LIBUV_SHA1=8d5131b6c1595920dd30644cd1435b4f344b46c8
+LIBUV_SHA1=52d72a52cc7ccd570929990f010ed16e2ec604c
```
Comment (tkelman) [inline deps/libuv.version]: why a partial sha? 52d72a52cc7ccd570929990f010ed16e2ec604c8

File: src/signals-unix.c [hunk 2]
```diff
@@ -513,10 +513,30 @@ static void *signal_listener(void *arg)
...
+    sigdelset(&sset, SIGABRT);
+    sigdelset(&sset, SIGSYS);
+    pthread_sigmask(SIG_SETMASK, &sset, 0);
+
     jl_sigsetset(&sset);
     while (1) {
         profile = 0;
-        sigwait(&sset, &sig);
+        sig = 0;
+        errno = 0;
```
Comment (yuyichao) [inline src/signals-unix.c]: This one shouldn't be necessary? Or is this just to help future debugging?

**Review by yuyichao — APPROVED** [review]

File: src/signals-unix.c (hunk 2 above)
Comment (vtjnash) [inline src/signals-unix.c]: future debugging

File: deps/checksums/libuv-52d72a52cc7ccd570929990f010ed16e2ec604c.tar.gz/md5 [hunk 3]
```diff
@@ -0,0 +1 @@
+0e6ec181c8bfd322a17dbd843b2b8ebb
```
Comment (tkelman) [inline deps/checksums/libuv-52d72a52cc7ccd570929990f010ed16e2ec604c.tar.gz/md5]: filenames are missing the last char

File: .travis.yml [hunk 4]
```diff
@@ -102,7 +102,7 @@ script:
     # capture the log, but only print it if `make deps` fails
     # try to show the end of the log first, because this log might be very long (> 4MB)
     # and thus be truncated by travis
-    - moreutils/mispipe "make $BUILDOPTS NO_GIT=1 -C deps 2> deps-err.log" "$BAR" > deps.log ||
+    - moreutils/mispipe "make \$BUILDOPTS NO_GIT=1 -C deps 2> deps-err.log" "$BAR" > deps.log ||
```
Comment (tkelman) [inline .travis.yml]: Please do not merge this unless you've verified that this branch builds from a completely clean cache. The deps should not need the spawn modification.

**Review by tkelman — CHANGES_REQUESTED** [review]

**Pull request closed without merging**